src/plugin/managers/extension-registry.spec.ts

```
@@ -0,0 +1,735 @@
+    });
+  });
+
+  describe('Extension Point Registration', () => {
+    it('should register all extension points from manifest', () => {
+      manager.registerExtensionPoints('test-plugin', mockManifest);
+      const registry = manager.getExtensionRegistry();
+
+      // Check routes
+      expect(registry.routes).toHaveLength(1);
+      expect(registry.routes[0]).toEqual({
+        ...mockManifest.extensionPoints!.routes![0],
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)